app/models/job_execution.rb

```diff
@@ -179,26 +201,27 @@ def lock_project(&block)
 
   class << self
     def find_by_id(id)
-      registry[id.to_i]
+      registry.find(id)
     end
 
-    def start_job(reference, job, &block)
-      new(reference, job).tap do |job_execution|
-        if enabled
-          registry[job.id] = job_execution
-          job_execution.start!(&block)
-          ActiveSupport::Notifications.instrument "job.threads", thread_count: registry.length
-        end
-      end
+    def active?(id, key: id)
+      registry.active?(key, id)
+    end
+
+    def queued?(id, key: id)
+      registry.queued?(key, id)
+    end
+
+    def start_job(reference, job, key: job.id, **options, &block)
+      registry.add(key, reference, job, options, &block)
     end
```

Comment:
Yeah, I agree. I just don't like. I made start_job take an execution instance instead, then on_complete blocks are still done on the instance and it's cleaner for JobQueue to know nothing about instantiating JobExecution.